### ISSUE TITLE
Element click penetration

### ISSUE THREAD

**Issue by xjy12345654** [slint-ui/slint#6812]
platform :windows11,slintPad
expected behavior：When self.checked=true and re1 overrides re2, clicking re1 should not trigger the callback fn of chekbox;
[](url)
code: [code_url](https://docs.slint.dev/latest/editor/?snippet=import+%7B+CheckBox+%7D+from+%22std-widgets.slint%22%3B%0A+%0Aexport+component+Recipe+inherits+Window+%7B%0A++++width%3A+200px%3B%0A++++height%3A+100px%3B%0A+++++re1+%3A%3D+Rectangle+%7B%0A++++++++property+%3Cduration%3E+ani_dutime%3A300ms%3B%0A++++++++x%3A+0%3B%0A++++++++y%3A+0%3B%0A++++++++width%3A+50px%3B%0A++++++++height%3A+50px%3B%0A++++++++background%3A+blueviolet%3B%0A++++++++z%3A+2%3B%0A++++++++animate+x+%7B+duration%3A+self.ani_dutime%3B+%7D%0A++++++++animate+y+%7B%0A++++++++++++delay%3A+self.ani_dutime%3B%0A++++++++++++duration%3A+self.ani_dutime%3B%0A++++++++%7D%0A++++%7D%0A%0A++++re2+%3A%3D+Rectangle+%7B%0A++++++++y%3A+re1.height%3B%0A++++++++background%3A+%23aaa%3B%0A++++++++VerticalLayout+%7B%0A++++++++++++CheckBox+%7B%0A++++++++++++++++text%3A+%22check%22%3B%0A++++++++++++++++toggled+%3D%3E+%7B%0A++++++++++++++++++++if+%28self.checked%29+%7B%0A+++++++++++++++++++++++%2F%2F+re1.x+%3D+root.width+-+re1.width%3B%0A++++++++++++++++++++++++re1.y+%3D+re1.height%3B%0A++++++++++++++++++++%7D+else+%7B%0A++++++++++++++++++++++++re1.x+%3D+0%3B%0A++++++++++++++++++++++++re1.y+%3D+0%3B%0A++++++++++++++++++++%7D%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)

```slint
import { CheckBox } from "std-widgets.slint";
export component Recipe inherits Window {
    width: 200px;
    height: 100px;
     re1 := Rectangle {
        property <duration> ani_dutime:300ms;
        x: 0;
        y: 0;
        width: 50px;
        height: 50px;
        background: blueviolet;
        z: 2;
        animate x { duration: self.ani_dutime; }
        animate y {
            delay: self.ani_dutime;
            duration: self.ani_dutime;
        }
    }

    re2 := Rectangle {
        y: re1.height;
        background: #aaa;
        VerticalLayout {
            CheckBox {
                text: "check";
                toggled => {
                    if (self.checked) {
                       // re1.x = root.width - re1.width;
                        re1.y = re1.height;
                    } else {
                        re1.x = 0;
                        re1.y = 0;
                    }
                }
            }
        }
    }
}
```

**Comment by hunger** [comment]
Interesting problem this, I do not think this is a bug though.

Rectangle are "transparent" to events, so I kind of expect the behavior this code exhibits.  You can put a `TouchArea { }` into `re1` and then the rectangle is no longer transparent for events.

**Comment by xjy12345654** [comment]
> 有趣的问题是，我不认为这是一个错误。
> 
> Rectangle 对事件是 “透明的”，所以我有点期待这段代码所展示的行为。您可以输入 into，然后矩形对于事件不再透明。`TouchArea { }``re1`

Thank you for your answer. Adding touch areas has indeed achieved the expected behavior
